lib/os/json.c

```diff
@@ -989,12 +1006,14 @@ static int encode(const struct json_obj_descr *descr, const void *val,
 				       ptr, append_bytes, data);
 	case JSON_TOK_NUMBER:
 		return num_encode(ptr, append_bytes, data);
-	case JSON_TOK_NULL:
-		return null_encode(ptr, append_bytes, data);
 	case JSON_TOK_FLOAT:
+		return float_encode(ptr, append_bytes, data);
+	case JSON_TOK_FLOAT_ASCII:
```

Comment:
This does not seem to be compatilbe. Are you sure no other code in zephyr does rely on `JSON_TOK_FLOAT` actually doing a `float_ascii_encode`?

Reply:
Or is this only be used internally?

Reply:
Yes, this is from the first change into the Zephyr v3.1 json library, which did not support floats yet. I will improve on it to be inline with the Zephyr v3.2 json library.